### ISSUE TITLE
Resolve T::Struct warning during test runs

### ISSUE THREAD

**Issue by maxveldink** [maxveldink/sorbet-schema#24]
While running the test suite, we see the following warnings a few times:

```
(eval at /Users/maxveldink/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/sorbet-runtime-0.5.11284/lib/types/props/has_lazily_specialized_methods.rb:65):1: warning: method redefined; discarding old __t_props_generated_deserialize
/Users/maxveldink/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/sorbet-runtime-0.5.11284/lib/types/props/has_lazily_specialized_methods.rb:91: warning: previous definition of __t_props_generated_deserialize was here
```

I'm not quite sure why we're redefining methods, but we should try and suppress this warning.

**Comment by maxveldink** [comment]
Fixed upstream in sorbet-runtime 0.5.11475 via sorbet/sorbet#7953, which wraps the lazy `class_eval` in `T::Configuration.without_ruby_warnings`. We already pin ≥ that version, and the suite (which runs with `-w` via `Minitest::TestTask`) is warning-free across seeds on both 0.5.11475 and 0.6.x. Separately, this gem no longer calls `T::Struct#serialize`/`.from_hash` at all since #102, so the lazy codegen path is never entered.
